src/Compilers/CSharp/Portable/Symbols/Synthesized/SynthesizedBackingFieldSymbol.cs

```diff
@@ -72,15 +77,165 @@ internal override bool HasPointerType
+        {
+            get
+            {
+                if (HasRuntimeSpecialName)
+                {
+                    return true;
+                }
+
+                var data = GetDecodedWellKnownAttributeData();
+                return data != null && data.HasSpecialNameAttribute;
+            }
+        }
+
+        /// <summary>
+        /// Returns data decoded from Obsolete attribute or null if there is no Obsolete attribute.
+        /// This property returns ObsoleteAttributeData.Uninitialized if attribute arguments haven't been decoded yet.
+        /// </summary>
+        internal sealed override ObsoleteAttributeData ObsoleteAttributeData
+        {
+            get
+            {
+                var containingType = (SourceMemberContainerTypeSymbol)_property.ContainingType;
+                if (!containingType.AnyMemberHasAttributes)
+                {
+                    return null;
+                }
+
+                var lazyCustomAttributesBag = _lazyCustomAttributesBag;
+                if (lazyCustomAttributesBag != null && lazyCustomAttributesBag.IsEarlyDecodedWellKnownAttributeDataComputed)
+                {
+                    var data = (CommonFieldEarlyWellKnownAttributeData)lazyCustomAttributesBag.EarlyDecodedWellKnownAttributeData;
+                    return data?.ObsoleteAttributeData;
+                }
+
+                return ObsoleteAttributeData.Uninitialized;
+            }
+        }
+
+        /// <summary>
+        /// Returns data decoded from well-known attributes applied to the symbol or null if there are no applied attributes.
+        /// </summary>
+        /// <remarks>
+        /// Forces binding and decoding of attributes.
+        /// </remarks>
+        private CommonFieldWellKnownAttributeData GetDecodedWellKnownAttributeData()
+        {
+            var attributesBag = _lazyCustomAttributesBag;
+            if (attributesBag == null || !attributesBag.IsDecodedWellKnownAttributeDataComputed)
+            {
+                attributesBag = this.GetAttributesBag();
+            }
+
+            return (CommonFieldWellKnownAttributeData)attributesBag.DecodedWellKnownAttributeData;
+        }
+
+        /// <summary>
+        /// Returns a bag of custom attributes applied on the backing field and data decoded from well-known attributes. Returns null if there are no attributes.
+        /// </summary>
+        /// <remarks>
+        /// Forces binding and decoding of attributes.
+        /// </remarks>
+        private CustomAttributesBag<CSharpAttributeData> GetAttributesBag()
+        {
+            var bag = _lazyCustomAttributesBag;
+            if (bag != null && bag.IsSealed)
+            {
+                return bag;
+            }
+
+            // We let the property drive and track completion
+            _ = LoadAndValidateAttributes(OneOrMany.Create(_property.CSharpSyntaxNode.AttributeLists), ref _lazyCustomAttributesBag, AttributeLocation.Field);
```

Comment:
>, AttributeLocation.Field [](start = 131, length = 25)

It feels like it shouldn't be required to explicitly pass the attribute location if IAttributeTargetSymbol.AttributesOwner is properly implemented.